x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/esnative/NativeRealmIntegTests.java

```diff
@@ -269,7 +270,24 @@ public void testAddAndGetRole() throws Exception {
         assertFalse("role should not exist after being deleted", resp2.hasRoles());
     }
 
-    public void testAddUserAndRoleThenAuth() throws Exception {
+    public void testAddUserAndRoleThenAuth() {
+        testAddUserAndRoleThenAuth("joe");
+    }
+
+    public void testAddUserWithInternalUsernameAndRoleThenAuth() {
+        testAddUserAndRoleThenAuth(randomInternalUsername());
+    }
+
+    public void testAuthWithInternalUsernameFailsWithoutCorrespondingUser() {
+        String token = basicAuthHeaderValue(randomInternalUsername(), new SecureString("s3krit-password"));
+        ElasticsearchSecurityException e = expectThrows(
+            ElasticsearchSecurityException.class,
+            () -> client().filterWithHeader(Collections.singletonMap("Authorization", token)).prepareSearch("idx").get()
+        );
+        assertThat(e.status(), is(RestStatus.UNAUTHORIZED));
+    }
+
+    public void testAddUserAndRoleThenAuth(String username) {
```

Comment:
```suggestion
    private void testAddUserAndRoleThenAuth(String username) {
```
? 
Otherwise, it shows up to be testable in IDE.

Reply:
Cheers, coulda sworn I made this private!